### PR TITLE
Added endpoint for retrieving available docker image names

### LINKED ISSUE #37: Create service/endpoints for hitting the docker registry and returning all image names
We need an endpoint in our services for retrieving all of the available containers.

### PR DESCRIPTION
**Pull request by ericjperry**
Added an ```/images``` endpoint which retrieves all available images from the configured docker registry, or runs a query against the registry if the 'query' parameter is provided.

Right now there's no default registry, and we could potentially make the docker hub the default registry, but it would involve a little bit more work. Happy to do that with this PR but @sapanshah previously said we didn't need to worry about that yet. This will also require credentials to our private docker registry, which I'll send to you guys via email.

@KrisSiegel @sapanshah 

Resolves #37

### CODE REVIEW

File: bakula/services/docker.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+# Setup authorization plugin
+token_secret = app.config.get('token_secret', 'password')
+auth_plugin = TokenAuthorizationPlugin(token_secret)
+app.install(auth_plugin)
+
+# Set up connection defaults
+auth = (app.config.get('registry.username'), app.config.get('registry.password'))
+registry_host = app.config.get('registry.host', 'registry.immuta.com')
+protocol = app.config.get('registry.protocol', 'https')
```
Comment (sapanshah) [inline bakula/services/docker.py]: Minor Nitpick, any reason not to just have a base url setting vs having two settings?

File: bakula/services/docker.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#   KIND, either express or implied.  See the License for the
+#   specific language governing permissions and limitations
+#   under the License.
+
+from bottle import Bottle, request
+from bakula.bottle import configuration
+from bakula.bottle.errorutils import create_error
+from bakula.security.tokenauthplugin import TokenAuthorizationPlugin
+import requests
```
Comment (sapanshah) [inline bakula/services/docker.py]: Add requests to the requirements.txt?

File: bakula/services/docker.py (hunk 1 above)
Comment (ericjperry) [inline bakula/services/docker.py]: So, I have to add the hostname to the image name if this is a private registry (when you call search on the api, the names returned do not have the hostname/registry name included). That could cause issues if Bakula is running somewhere where it has access to the internet and the docker hub has an image with the same name. I could just have the full URL but I'd have to parse the hostname out (wouldn't be hard), I just got lazy. Can do it if you'd like.

Comment (sapanshah) [inline bakula/services/docker.py]: Thats fine, its a minor nitpick.

**Comment by KrisSiegel** [comment]
Other than Soup's comment about updating requirements.txt this seems good to me :+1:

**Pull request merged**